### PR TITLE
Silencing invalid error messages being logged with Template Expressions

### PR DESCRIPTION
**Pull request by blnicho**
## Summary/Motivation:
One unintended side effect of the new expression system is that error messages are now being logged every time Template Expressions are used. This PR defers error handling and error message logging to the Template Expression system.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/core/expr/numvalue.py [hunk 1]
```diff
@@ -217,13 +217,23 @@ def value(obj, exception=True):
         #
         # Here, we try to catch the exception
         #
+
+        # This import needs to be deferred to here due to circular
+        # imports
+        from pyomo.core.expr.current import TemplateExpressionError
```
Comment (jsiirola) [inline pyomo/core/expr/numvalue.py]: Does it break the circular reference to move the TemplateExpressionError definition either into this file, or better yet into a separate file?

**Comment by michaelbynum** [comment]
Oops. I did not realize John reviewed this already...

File: pyomo/core/expr/numvalue.py (hunk 1 above)
Comment (blnicho) [reply pyomo/core/expr/numvalue.py]: Yes, moving TemplateExpressionError to pyomo/core/expr/numvalue.py breaks the circular reference. I also tried moving it to pyomo/core/base/template_expr.py and pyomo/core/base/misc.py but ran into the same circular reference problem. My only hesitation with making this change is that numvalue.py seems like a weird place for this error definition.

Comment (qtothec) [reply pyomo/core/expr/numvalue.py]: Yeah, I agree that `numvalue.py` seems like a really odd place to put it. Why not put it in an `expr_errors.py` file? Or an `expr_util.py` file?

Comment (blnicho) [reply pyomo/core/expr/numvalue.py]: @qtothec yeah, I like that idea better

**Pull request merged**